packages/job-worker/src/playout/adlib.ts

```diff
@@ -109,10 +109,29 @@ export async function takePieceAsAdlibNow(context: JobContext, data: TakePieceAs
 						)
 						break
 					case IBlueprintDirectPlayType.AdLibAction: {
+						const playlist = cache.PlaylistId
 						const executeProps = pieceToCopy.allowDirectPlay
+						const showStyle = await context.getShowStyleCompound(
+							rundown.showStyleVariantId,
+							rundown.showStyleBaseId
+						)
+						const blueprint = await context.getShowStyleBlueprint(showStyle._id)
+
+						const currentPartInstance = playlist.currentPartInstanceId
+							? await context.directCollections.PartInstances.findOne(playlist.currentPartInstanceId)
```

Comment:
This should be pulling from the cache, not the database.

In fact, we already have the currentPartInstance as a variable (code might want some refactoring to clarify this) on line 86. We confirmed on line 63 that it is the currentPartInstance.